docusaurus-plugin-moonwave/src/index.js

```diff
@@ -86,6 +94,77 @@ function parseClassOrder(content, classOrder, nameSet) {
   }
 }
 
+function parseSimpleApiCategories(luaClass, apiCategories) {
+  const tagSet = new Set(
+    luaClass.functions.filter((func) => func.tags).flatMap((func) => func.tags)
+  )
+
+  let listedCategories = []
+  apiCategories.forEach((category) => {
+    if (tagSet.has(category)) {
+      listedCategories.push({
+        value: capitalize(category),
+        id: category,
+        children: luaClass.functions
+          .filter((func) => func.tags && func.tags.includes(category))
+          .map((member) => {
+            return {
+              value: addFunctionTypeSymbol(member.name, member.function_type),
+              id: member.name,
+              children: [],
+            }
+          })
+          .sort((childA, childB) => childA.id.localeCompare(childB.id)),
+      })
+    }
+  })
+
+  const baseCategories = SECTIONS.map((section) => ({
+    value: capitalize(section),
+    id: section,
+    children: luaClass[section]
+      .filter(
+        (member) =>
+          !member.tags ||
+          member.tags.some((tag) => !apiCategories.includes(tag))
+      )
+      .map((member) => ({
+        value: addFunctionTypeSymbol(member.name, member.function_type),
+        id: member.name,
+        children: [],
+      }))
+      .sort((childA, childB) => childA.id.localeCompare(childB.id)),
+  }))
+
+  return [...listedCategories, ...baseCategories]
+}
+
+function parseBaseApiCategories(luaClass) {
+  const baseCategories = SECTIONS.map((section) => ({
+    value: capitalize(section),
+    id: section,
+    children: luaClass[section]
+      .map((member) => ({
+        value: addFunctionTypeSymbol(member.name, member.function_type),
+        id: member.name,
+        children: [],
+      }))
+      .sort((childA, childB) => childA.id.localeCompare(childB.id)),
+  }))
+
+  return baseCategories
+}
+
+function parseApiCategories(luaClass, apiCategories) {
+  if (typeof apiCategories[0] === "string") {
+    // Handles simple apiCategories array assignment
+    return parseSimpleApiCategories(luaClass, apiCategories)
+  } else {
+    // Handles where no apiCategory config is provided
+    return parseBaseApiCategories(luaClass)
+  }
+}
+
```

Comment:
Why do we branch here? The code in `parseSimpleApiCategories` should work just fine if it receives an empty array. I think it's safe to remove both of these functions and just rename `parseSimpleApiCategories` to `parseApiCategories`